tests/basilisp/contrib/nrepl_server_test.lpy

```diff
@@ -0,0 +1,598 @@
+(ns tests.basilisp.contrib.nrepl-server-test
+  (:require
+   [basilisp.contrib.bencode :as bc]
+   [basilisp.io :as bio]
+   [basilisp.contrib.nrepl-server :as nr]
+   [basilisp.string :as str :refer [starts-with?]]
+   [basilisp.test :refer [deftest are is testing]])
+  (:import
+   os
+   socket
+   tempfile
+   threading
+   time))
+
+(def ^:dynamic *nrepl-port*
+  "The port the lpy:py:`with-server` is bound to."
+  nil)
+
+(defmacro with-server
+  [opts & body]
+  "Create an nREPL server on a thread with
+  lpy:fn:`basilisp.nrepl-server/server-make` passing in ``opts``, bind
+  its port to `*nrepl-port*`, run ``body`` on the main thread, and
+  then shutdown server."
+  `(let [srv# (nr/server-make ~opts)]
+     (doto (threading/Thread
+            ~'**
+            :target (fn []
+                      (.serve-forever srv#))
+            :daemon true)
+       (.start))
+     (try
+       (binding [*nrepl-port* (second (.-server-address srv#))]
+         ~@body)
+       (finally
+         (doto srv#
+           (.shutdown)
+           (.server-close))))))
+
+(defmacro with-connect [client & body]
+  "Open up a connection to the nREPL-server at ``*nrepl-port*`` and
+  run ``body``, with the ``client`` exposed as an anaphoric binding.
```

Comment:
I'm not familiar with this term `anaphoric` -- what does this mean here?

Reply:
Anophoric macros use anaphora (in our case, a single one, the value of `client`) in the body, to refer to forms/computations that it would be not be visible to the user otherwise. The user can use that value (a symbol) to refer to the computation the body.

In our case, whatever symbol the user passes in for `client` it can be used in the body to refer to the client connection.

https://letoverlambda.com/index.cl/guest/chap6.html

Reply:
Interesting. Thanks for the explanation!